### DISCUSSION TITLE
How to make type comparisons?

### DISCUSSION THREAD

**Discussion by danvinci** [cjdoris/PythonCall.jl#259] [Q&A] [answered]
Hello!

First of all, thanks for this amazing package - it's saving me headaches!

I've got a question concerning how to make Python-types comparisons from Julia:

I'm using pymongo via PythonCall for querying a MongoDB instance (Mongoc.jl doesn't work as expected), and when mapping back the query results to a Julia Dict, ObjectIds are left behind (all good, it makes sense).

Being the only kind of value that has a "Py" type left in my dictionary, I can simply call 'pystr' + 'pyconvert' for retrieving the ID as a Julia String.

What I'd like, is to make a type check before performing this operation, knowing that in Python their type is: 
`<class 'bson.objectid.ObjectId'>`

Calling pytype on the same object I get: 
`Python type: <class 'bson.objectid.ObjectId'>`

Does anyone encountered a similar problem before?

**Comment by cjdoris** [comment, marked as answer]
What's the actual question? There is `pyissubclass` and `pyisinstance` which you can use to check types.

**Reply by danvinci** [reply]
Thank you!

My question was about what goes in the function call.

In Julia (with Julia types) I can do:
`isa(x, String)` 

Using PythonCall I'd have to do:
`pyisinstance(x, <python type>)`

But what I don't know (maybe it's a stupid question) is how to pass the <python type>::Type argument to the function.

I can retrieve an example of variable of the type that I need to make the comparison against, retrieve its type via pytype, and using this "type" as argument - but wondering if there's a more straightforward way knowing the name of the type.

**Reply by cjdoris** [reply]
Oh I see. You'd get the type just as you would in Python, by importing it from some module. For your particular case the type is `pyimport("bson.objectid").ObjectId`.

**Reply by danvinci** [reply]
Thank you, that worked!!
